src/rust/cryptography-x509/src/extensions.rs

```diff
@@ -231,7 +231,32 @@ pub struct BasicConstraints {
 
 pub type SubjectAlternativeName<'a> = asn1::SequenceOf<'a, name::GeneralName<'a>>;
 pub type IssuerAlternativeName<'a> = asn1::SequenceOf<'a, name::GeneralName<'a>>;
-pub type ExtendedKeyUsage<'a> = asn1::SequenceOf<'a, asn1::ObjectIdentifier>;
+
+pub struct ExtendedKeyUsage<'a>(asn1::SequenceOf<'a, asn1::ObjectIdentifier>);
+
+impl<'a> Iterator for ExtendedKeyUsage<'a> {
+    type Item = asn1::ObjectIdentifier;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.0.next()
+    }
+}
+
+impl<'a> asn1::SimpleAsn1Readable<'a> for ExtendedKeyUsage<'a> {
```

Comment:
Do we need a rust-asn1 builtin way to enforce size constraints? Needing to write this for everything feels painful

Reply:
Yeah, that would be fantastic. Do you want me to hold off on this and look at generalizing?

Reply:
(One idea: there could be a `NonemptySequenceOf` or similar.)

Reply:
I think it'd be good to at least figure out what our options are for doing this in rust-asn1. Maybe it'll be more trouble than it's worth, but this is far from the only place for this.

Reply:
Gotcha -- for this kind of pattern in particular, some options I can think of are:

1. Retain the `pub type = ...` pattern, but add a new set of constraining `rust-asn1` types (e.g. `NonemptySequenceOf`, `NonemptySetOf`). 
2. Prefer the newtype pattern, and offer additional proc-macro annotation support, e.g.:

```rust
struct Whatever<'a>(#[nonempty] asn1::SequenceOf<'a, asn1::ObjectIdentifier>);
```

I personally find (2) a little bit cleaner, but it's probably slightly less trivial to implement 🙂

Reply:
Another idea: make it `struct SequenceOf<T, const MINIMUM_SIZE: usize = 0, const MAXIMUM_SIZE: usize = usize::MAX>`

Reply:
That seems a lot nicer!

Reply:
Yeah, I kind of like it, my concerns are:

1. Will it break type inferences or otherwise cause pain for users who don't override min/max size?
2. This means you'll no longer be able to do anything with an instance with different bounds, even if it's already parsed.
3. Does it have incorrect generality? (size bounds are really applicable to a whole bunch of types, do we have to repeat these everywhere?)

Reply:
Yeah, hmm. I can play around with it a bit and see if I can answer those.

Reply:
Dope. If it looks viable, feel free to go directly to rust-asn1 PR

Reply:
Opened https://github.com/alex/rust-asn1/pull/470 as a WIP for the above; I'll try it against `cryptography-x509` tonight if I have the time 🙂